### PR TITLE
Fix #1289 Added option to add photos to favourites table in the realm database.

### LINKED ISSUE #1289: No feature to add photos to favourite collection.
**Actual Behaviour**

Currently there is no option to add photos to users favourite collection.

**Expected Behaviour**

Option to add photos to favourite collection should be there. It will help users to keep track of important photos and so they would not need to go through every photo to find important ones. A favourites folders will be there to display all the favourite marked photos.

**Steps to reproduce it**

Click on any photo, there is no option to add it to favourites.

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by sauravvishal8797**
First step for issue #1289   

Changes: Upon clicking on the Add to favourites option in the menu, the photo uri is stored in the realm database.

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    }
+
+    public String getDescription(){
+        return description;
+    }
+
+}
+
+
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java]: Avoid unnecessary new lines

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java [hunk 2]
```diff
@@ -537,13 +538,43 @@ public void folderSelected(String path) {
...
+                        realpath);
+                if(query.count() == 0){
+
+                    realm.beginTransaction();
+                    fav = realm.createObject(FavouriteImagesModel.class,
+                            realpath);
+                    ImageDescModel q = realm.where(ImageDescModel.class).equalTo("path", realpath).findFirst();
+                    if(q != null) {
+                        fav.setDescription(q.getTitle());
+
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Here also

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java [hunk 3]
```diff
@@ -537,13 +538,43 @@ public void folderSelected(String path) {
...
+                    }
+                    else{
+                        fav.setDescription(" ");
+                    }
+                    realm.commitTransaction();
+                    Toast.makeText(this, "Image is succesfully added in the favourites list", Toast.LENGTH_SHORT).show();
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Use snackbar instead

Comment (sauravvishal8797) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Yeah. Will do the change.

**Comment by mohitmanuja** [comment]
Please do the suggested changes.

**Comment by mohitmanuja** [comment]
@sauravvishal8797

**Comment by anantprsd5** [comment]
@sauravvishal8797 Resolve conflicts.

**Comment by sauravvishal8797** [comment]
@anantprsd5 @mohitmanuja Done. Resolved the conflicts.

File: app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public void setDescription(String description){
+        this.description = description;
+    }
+
+    public String getDescription(){
+        return description;
+    }
+}
+
+
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java]: Avoid new lines. Resolve conflicts

**Comment by sauravvishal8797** [comment]
@anantprsd5 @mohitmanuja   made the changes. Please review.

File: app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public void setDescription(String description){
+        this.description = description;
+    }
+
+    public String getDescription(){
+        return description;
+    }
+}
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/data/local/FavouriteImagesModel.java]: No new line at the end of file

**Pull request merged**